### ISSUE TITLE
error: No such file or directory

### ISSUE THREAD

**Issue by ghost** [Luminarys/synapse#135]
synapse at https://github.com/Luminarys/synapse/commit/4a6e6c33b4c36eca89d216906d615797ba9a519e

I have torrent with four files, only one is selected for download via receptor (`fileA`).
Shortly after start, I get an error:
```
06/02/19 14:24:08 [synapse::disk:129] D: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```
After some digging, I can report this:
- Not founded file is `fileC`, which priority is set to `0`
- [write_piece](https://github.com/Luminarys/synapse/blob/HEAD@%7B2019-06-02T12:07:44Z%7D/src/torrent/mod.rs#L1672)
want to write piece at `fileC` somehow (I think some part of that piece is for `fileA`, but it starts at `fileC`, is that correct?)
- [loc.allocate](https://github.com/Luminarys/synapse/blob/HEAD@%7B2019-06-02T12:07:44Z%7D/src/disk/job.rs#L329) is false, due to zero priority.
- [ensure_exists](https://github.com/Luminarys/synapse/blob/HEAD@%7B2019-06-02T12:07:44Z%7D/src/disk/cache.rs#L164) is invoked with size=None, which leads to
- [create(false)](https://github.com/Luminarys/synapse/blob/HEAD@%7B2019-06-02T12:07:44Z%7D/src/disk/cache.rs#L230-L234)
- and as we do not have `fileC` created, this triggers NotFound error;

**Comment by Luminarys** [comment]
Thanks for the detailed info. This does seem to be a rather nasty bug around 0 priority files, I'll think about some way of dealing with it.

**Comment by Luminarys** [comment]
This should be fixed with 53b986c but if it's still not please reopen the issue.

**Comment by ghost** [comment]
Works fine, thank you!

**Comment by Luminarys** [comment]
This issue should be fixed, but do make sure you're on at least 849c4842a52c23461f2f896ccdd7ee21bee815c1 since I accidentally added a few disk read issues in the process of fixing this.

**Comment by ghost** [comment]
Copy that
